app/main/posts/views/filters/filters-dropdown.directive.js

```diff
@@ -1,77 +1,75 @@
 module.exports = FiltersDropdown;
 
-FiltersDropdown.$inject = ['PostFilters', 'ModalService', '$rootScope', '_', '$location', 'SavedSearchEndpoint'];
-function FiltersDropdown(PostFilters, ModalService, $rootScope, _, $location, SavedSearchEndpoint) {
+FiltersDropdown.$inject = [];
+function FiltersDropdown() {
     return {
         restrict: 'E',
-        require: 'ngModel',
         scope: {
             dropdownStatus: '=',
-            applyFilters: '=',
-            filtersVar: '=',
-            cancel: '='
+            filters: '='
         },
-        link: FiltersDropdownLink,
+        controller: FiltersDropdownController,
         template: require('./filters-dropdown.html')
     };
-    function FiltersDropdownLink($scope, $element, $attrs, ngModel) {
-        $scope.canUpdateSavedSearch;
-        PostFilters.reactiveFilters = false;
-        $scope.$watch(PostFilters.getModeId, function (newValue, oldValue) {
-            if (oldValue !== newValue || typeof ($scope.canUpdateSavedSearch) === 'undefined') {
-                setSavedSearchUpdateStatus();
-            }
-        });
-        // Init an empty saved search
-        $scope.savedSearch = {
-            view : 'map',
-            role : []
-        };
-        // Check if we can edit
-        function setSavedSearchUpdateStatus() {
-            var savedSearchId = PostFilters.getModeId();
-            if (savedSearchId) {
-                SavedSearchEndpoint.get({id: savedSearchId}, function (savedSearch) {
-                    $scope.canUpdateSavedSearch = _.contains(savedSearch.allowed_privileges, 'update');
-                });
-            }
-        }
+}
 
-        $scope.applyFiltersLocked = function () {
-            PostFilters.reactiveFilters = true;
-            $scope.dropdownStatus.isopen = !$scope.dropdownStatus.isopen;
-        };
-        $scope.clearFilters = function () {
-            if (PostFilters.getMode() === 'savedsearch' && PostFilters.getModeId()) {
-                PostFilters.setMode('all');
-                var viewParam = $scope.$resolve.$transition$.params().view ? $scope.$resolve.$transition$.params().view : 'data';
-                $location.url('/views/' + viewParam);
-            }
-            $scope.filtersVar = PostFilters.clearFilters();
-            $scope.dropdownStatus.isopen = !$scope.dropdownStatus.isopen;
-            PostFilters.reactiveFilters = true;
-            $scope.canUpdateSavedSearch = false;
+FiltersDropdownController.$inject = ['$scope', '$state', 'PostFilters', 'ModalService', '$rootScope', '_', '$location', 'SavedSearchEndpoint'];
+function FiltersDropdownController($scope, $state, PostFilters, ModalService, $rootScope, _, $location, SavedSearchEndpoint) {
```

Comment:
@rowasc so I converted this to a controller simply so I could inject $state here. 
Unfortunately the indent change makes the diff huge :/